usecases/modules/compare.go

```diff
@@ -0,0 +1,137 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package modules
+
+import (
+	"context"
+
+	"github.com/weaviate/weaviate/entities/additional"
+	"github.com/weaviate/weaviate/entities/models"
+	"github.com/weaviate/weaviate/entities/modulecapabilities"
+	"github.com/weaviate/weaviate/entities/moduletools"
+	"github.com/weaviate/weaviate/entities/schema"
+	"github.com/weaviate/weaviate/entities/search"
+)
+
+func reVectorize(ctx context.Context, cfg moduletools.ClassConfig, mod modulecapabilities.Vectorizer, object *models.Object, class *models.Class, sourceProperties []string, targetVector string, findObjectFn modulecapabilities.FindObjectFn) (bool, models.AdditionalProperties, []float32) {
+	textProps, mediaProps, err := mod.VectorizedProperties(cfg)
+	if err != nil {
+		return true, nil, nil
+	}
+
+	type compareProps struct {
+		Name    string
+		IsArray bool
+	}
+	propsToCmpare := make([]compareProps, 0)
+
+	for _, prop := range class.Properties {
+		if len(prop.DataType) > 1 {
+			continue // multi cref
+		}
+
+		// for named vectors with explicit source properties, skip if not in the list
+		if sourceProperties != nil {
+			found := false
+			for _, sourceProp := range sourceProperties {
```

Comment:
`sourceProperties` can be converted to map upfront for easier presence check to avoid going through it for each of class properties.